parsers/html.c

```diff
@@ -8,35 +8,434 @@
+				token->type = TOKEN_OTHER;
+			else if (d == '/')
+				token->type = TOKEN_TAG_START2;
+			else
+			{
+				ungetcToInputFile (d);
+				token->type = TOKEN_TAG_START;
+			}
+			break;
+		}
+		case '/':
+		{
+			int d = getcFromInputFile ();
+			if (d == '>')
+				token->type = TOKEN_TAG_END2;
+			else
+			{
+				ungetcToInputFile (d);
+				token->type = TOKEN_OTHER;
+			}
+			break;
+		}
+		case '>':
+		{
+			token->type = TOKEN_TAG_END;
+			break;
+		}
+
+		case '=':
+			token->type = TOKEN_EQUAL;
+			break;
+
+		case '"':
+		case '\'':
+		{
+			const int delimiter = c;
+			c = getcFromInputFile ();
+			while (c != EOF && c != delimiter)
+			{
+				vStringPut (token->string, c);
+				c = getcFromInputFile ();
+			}
+			token->type = TOKEN_STRING;
+			break;
+		}
+
+		default:
+		{
+			do
+			{
+				vStringPut (token->string, tolower (c));
+				c = getcFromInputFile ();
+			}
+			while (c != EOF && c != ' ' && c != '\t' && c != '<' && c != '/' &&
+				   c != '=' && c != '>' && c != '\'' && c != '"');
+			if (c != EOF)
+				ungetcToInputFile (c);
+			token->type = TOKEN_NAME;
+			break;
+		}
+	}
+}
+
+static void readTagContent (tokenInfo *token, vString *text, boolean isScript)
+{
+	tokenType type;
+
+	readTokenText (token, text != NULL, isScript);
+	if (text != NULL)
+		vStringCat (text, token->string);
+
+	do
+	{
+		readToken (token, FALSE);
+		type = token->type;
+		if (type == TOKEN_COMMENT)
+		{
+			readTokenText (token, text != NULL, isScript);
+			if (text != NULL)
+				vStringCat (text, token->string);
+		}
+		else if (type == TOKEN_TAG_START)
+		{
+			readTag (token, text);
+			readTokenText (token, text != NULL, isScript);
+			if (text != NULL)
+				vStringCat (text, token->string);
+		}
+	}
+	while (type == TOKEN_COMMENT || type == TOKEN_TAG_START);
+}
+
+static void readTag (tokenInfo *token, vString *text)
+{
+	boolean textCreated = FALSE;
+	readToken (token, TRUE);
+	if (token->type == TOKEN_NAME)
+	{
+		keywordId startTag;
+
+		startTag = lookupKeyword (vStringValue (token->string), Lang_html);
+		if (text == NULL &&
+			(startTag == KEYWORD_h1 || startTag == KEYWORD_h2 || startTag == KEYWORD_h3))
+		{
+			text = vStringNew ();
+			textCreated = TRUE;
+		}
+
+		do
+		{
+			readToken (token, TRUE);
+			if (startTag == KEYWORD_a && token->type == TOKEN_NAME)
+			{
+				keywordId attribute = lookupKeyword (vStringValue (token->string), Lang_html);
+
+				if (attribute == KEYWORD_name)
+				{
+					readToken (token, TRUE);
+					if (token->type == TOKEN_EQUAL)
+					{
+						readToken (token, TRUE);
+						if (token->type == TOKEN_STRING || token->type == TOKEN_NAME)
+							makeSimpleTag (token->string, HtmlKinds, K_ANCHOR);
+					}
+				}
+			}
+		}
+		while (token->type != TOKEN_TAG_END && token->type != TOKEN_TAG_END2 &&
+			   token->type != TOKEN_EOF);
+
+		if (token->type == TOKEN_TAG_END)
+		{
+			readTagContent (token, text, startTag == KEYWORD_script);
+
+			readToken (token, TRUE);
+			if (textCreated && text != NULL)
```

Comment:
Yeah ok, but my point is that this refers to [L347](https://github.com/universal-ctags/ctags/pull/1079/files/3048c15bf2917579a3e81e0cfff8d1ca3ad52ea7#diff-e619353ef58a45cb65c0674de4ebdb9fR347) in a really not obvious way, which made me wonder at first if the heading kind selection below couldn't get fooled and generate `H3` tags for a `A` tag.

If it was something like `isHeading` or maybe simply `/* if we were reading a heading */`, it would be a lot clearer to the casual reader.